core/tests/org.fusesource.ide.camel.tests.util/src/org/fusesource/ide/camel/tests/util/MavenProjectHelper.java

```diff
@@ -48,34 +48,49 @@
 public class MavenProjectHelper {
 
 
-	public IProject[] importProjects(File src, String[] pomNames) throws IOException, CoreException {
+	private static final class MarkerComparatorByLine implements Comparator<IMarker> {
+		@Override
+		public int compare(IMarker o1, IMarker o2) {
+			int lineNumber1 = o1.getAttribute(IMarker.LINE_NUMBER, -1);
+			int lineNumber2 = o2.getAttribute(IMarker.LINE_NUMBER, -1);
+			if(lineNumber1 < lineNumber2) {
+				return -1;
+			}
+			if(lineNumber1 > lineNumber2) {
+				return 1;
+			}
+			// Markers on the same line
+			String message1 = o1.getAttribute(IMarker.MESSAGE, "");
+			String message2 = o2.getAttribute(IMarker.MESSAGE, "");
+			return message1.compareTo(message2);
+		}
+	}
+
+	public IProject[] importProjects(File src, String[] pomNames) throws CoreException, IOException {
```

Comment:
I only moved the class.
This is used in tests only so maybe I should modify to throw the generic Exception? (even if it will reported as a defect too by Sonar)